src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -199,6 +199,12 @@ def upgrade_start(self, image: str, version: str) -> str:
                 self._save_upgrade_state()
                 return 'Resumed upgrade to %s' % self.target_image
             return 'Upgrade to %s in progress' % self.target_image
+        if len([daemon for daemon in self.mgr.cache.get_daemons_by_type('mgr')
+                if daemon.status == DaemonDescriptionStatus.running]) < 2:
+            raise OrchestratorError('Need at least 2 running mgr daemons for upgrade')
+        if len([daemon for daemon in self.mgr.cache.get_daemons_by_type('mon')
+                if daemon.status == DaemonDescriptionStatus.running]) < 3:
+            raise OrchestratorError('Need at least 3 running mon daemons for upgrade')
```

Comment:
```suggestion
        if len([daemon for daemon in self.mgr.cache.get_daemons_by_type('mgr')
                if daemon.status == DaemonDescriptionStatus.running]) < 2:
            raise OrchestratorError('Need at least 2 running mgr daemons for upgrade')
        if len([daemon for daemon in self.mgr.cache.get_daemons_by_type('mon')
                if daemon.status == DaemonDescriptionStatus.running]) < 3:
            raise OrchestratorError('Need at least 3 running mon daemons for upgrade')
```
(nit) at first glance I thought the indentation was off here, maybe we can improve the readability ?

Reply:
I split up these lines a bit. Should be more readable now.